compiler/src/iree/compiler/GlobalOptimization/RaiseSpecialOps.cpp

```diff
@@ -679,6 +679,79 @@ static Value rewriteCatNegateAndSlice(RewriterBase &rewriter,
   return createCatNegateAndSlice(rewriter, outTensor, source);
 }
 
+// Match an extract slice that has a shape of size N and an output shape of size
+// N-1, where the extraction happened along only the innermost dimension.
+//
+// e.g.
+// %extracted_slice = tensor.extract_slice
+//   %1[0, 0, 0, 1] [1024, 7, 7, 1] [1, 1, 1, 1]
+//   : tensor<1024x7x7x2xf32> to tensor<1024x7x7xf32>
+// Rewrite by adding a transpose that makes dim N the innermost and slice along
```

Comment:
did you mean outermost here? dim N is moving to dim 0, correct?

Reply:
oops, yes you're correct